guardian/mobile-apps-api-models
# Adopt GHA Scala Library Release Workflow

## What does this change?

This change migrates the release workflow of mobile-apps-api-models to a new [reusable release workflow](https://github.com/guardian/gha-scala-library-release-workflow). The reusable workflow is a neater solution that what's there currently because of the way it handles [credentials](https://github.com/guardian/gha-scala-library-release-workflow/blob/add-configuration-guidence/docs/credentials.md) and it's [security design](https://github.com/guardian/gha-scala-library-release-workflow/blob/add-configuration-guidence/docs/security-design.md).

The use of the reusable workflow will change the release process a little:
- the release won't be created by humans via the github ui (the workflow generates and publishes the release) - this means that the version is determined by a version.sbt file, not by us defining the version as a tag in the release UI
- at the moment the reusable workflow [doesn't support snapshot releases](https://github.com/guardian/gha-scala-library-release-workflow/issues/10) - this means we'll only be able to make production releases with the new release. We can always uses the sbt command `publishLocal` to test/verify local changes.

In this companion PR we grant this repo access to the necessary secrets required by the reusable workflow: https://github.com/guardian/github-secret-access/pull/25

After merging this PR I will create a release and verify the action completes successfully. If there's an issue we will fix forward.

After confirming the new release workflow executes successfully we will delete the unused secrets from the repo (PGP_SECRET, PASSPHRASE, SONATYPE_USERNAME, SONATYPE_PASSWORD)

## How to test



After this change is deployed we should be able to:
- trigger the new `Release` workflow via the `Actions` tab
- verify the action completes successfully
- [find](https://index.scala-lang.org/guardian/mobile-apps-api-models/artifacts/mobile-apps-api-models-v0) the next version of the library on Maven

## How can we measure success?



The old workflow is no longer needed and we can release successfully using the new one.

## Have we considered potential risks?



We can no longer test changes via CI - instead we'll have to release changes to test e.g. in CODE. I think the benefits the reusable workflow brings is worth this compromise.
